### PR TITLE
Created folder for losses in Machine_Learning

### PR DESCRIPTION
**Pull request by THEGAMECHANGER416**
### Describe your change:
Added a directory named `losses` in `machine_learning` that contains two loss functions namely `binary_cross_entropy.py` and `mean_squared_error.py`

* [x] Add an algorithm?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by THEGAMECHANGER416** [comment]
@darkstar Can you please approve this P.R for hacktoberfest ?
P.S. Sorry for all the messages, the code now runs perfectly and passes all tests. 
Thanks

File: machine_learning/losses/binary_cross_entropy.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    # Clip predicted probabilities to avoid log(0) and log(1)
+    y_pred = np.clip(y_pred, epsilon, 1 - epsilon)
+
+    # Calculate binary cross-entropy loss
+    bce_loss = -(y_true * np.log(y_pred) + (1 - y_true) * np.log(1 - y_pred))
+
+    # Take the mean over all samples
+    bce_loss = np.mean(bce_loss)
+
+    return bce_loss
```
Comment (cclauss) [inline machine_learning/losses/binary_cross_entropy.py:34]:
```suggestion
    return np.mean(bce_loss)
```
https://docs.astral.sh/ruff/rules/unnecessary-assign/

File: machine_learning/losses/mean_squared_error.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    >>> mse
+    0.028000000000000032
+    """
+    if len(y_true) != len(y_pred):
+        raise ValueError("Input arrays must have the same length.")
+
+    squared_errors = np.square(np.subtract(y_true, y_pred))
+    mse = np.mean(squared_errors)
+
+    return mse
```
Comment (cclauss) [inline machine_learning/losses/mean_squared_error.py:28]: https://docs.astral.sh/ruff/rules/unnecessary-assign/

File: machine_learning/losses/mean_squared_error.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    - y_pred: The predicted values.
+
+    Returns:
+    - mse: The Mean Squared Error between y_true and y_pred.
+
+    Example usage:
+    >>> true_values = np.array([1.0, 2.0, 3.0, 4.0, 5.0])
+    >>> predicted_values = np.array([0.8, 2.1, 2.9, 4.2, 5.2])
+    >>> mse = mean_squared_error(true_values, predicted_values)
+    >>> mse
```
Comment (cclauss) [inline machine_learning/losses/mean_squared_error.py:19]:
```suggestion
    >>> mean_squared_error(true_values, predicted_values)
```

File: machine_learning/losses/binary_cross_entropy.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    - epsilon: Small constant to avoid numerical instability.
+
+    Returns:
+    - bce_loss: Binary Cross-Entropy Loss.
+
+    Example Usage:
+    >>> true_labels = np.array([0, 1, 1, 0, 1])
+    >>> predicted_probs = np.array([0.2, 0.7, 0.9, 0.3, 0.8])
+    >>> bce_loss = binary_cross_entropy(true_labels, predicted_probs)
+    >>> bce_loss
```
Comment (cclauss) [inline machine_learning/losses/binary_cross_entropy.py:22]:
```suggestion
    >>> binary_cross_entropy(true_labels, predicted_probs)
```

File: machine_learning/losses/mean_squared_error.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    >>> true_values = np.array([1.0, 2.0, 3.0, 4.0, 5.0])
+    >>> predicted_values = np.array([0.8, 2.1, 2.9, 4.2, 5.2])
+    >>> mean_squared_error(true_values, predicted_values)
+    0.028000000000000032
+    """
+    if len(y_true) != len(y_pred):
+        raise ValueError("Input arrays must have the same length.")
+
+    squared_errors = np.square(np.subtract(y_true, y_pred))
+    return np.mean(squared_errors)
```
Comment (tianyizheng02) [inline machine_learning/losses/mean_squared_error.py:25]:
```suggestion
    squared_errors = (y_true - y_pred) ** 2
    return np.mean(squared_errors)
```
You can just use Python operators on numpy ndarrays

File: machine_learning/losses/mean_squared_error.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import numpy as np
```
Comment (tianyizheng02) [inline machine_learning/losses/mean_squared_error.py:1]: Please include an explanation of MSE (along with a link to a source as well as the formula) in a header docstring

File: machine_learning/losses/binary_cross_entropy.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import numpy as np
```
Comment (tianyizheng02) [inline machine_learning/losses/binary_cross_entropy.py:1]: Please include an explanation of BSE (along with a link to a source as well as the formula) in a header docstring

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
Could we also name the directory `loss_functions` rather than `losses`? I feel that it's more accurate.

**Comment by THEGAMECHANGER416** [comment]
@tianyizheng02 Done all the requested changes
Thanks

File: machine_learning/loss_functions/mean_squared_error.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    - mse: The Mean Squared Error between y_true and y_pred.
+
+    Example usage:
+    >>> true_values = np.array([1.0, 2.0, 3.0, 4.0, 5.0])
+    >>> predicted_values = np.array([0.8, 2.1, 2.9, 4.2, 5.2])
+    >>> mean_squared_error(true_values, predicted_values)
+    0.028000000000000032
+    """
+    if len(y_true) != len(y_pred):
+        raise ValueError("Input arrays must have the same length.")
```
Comment (tianyizheng02) [inline machine_learning/loss_functions/mean_squared_error.py:36]: Can we have a doctest that tests this case?

File: machine_learning/loss_functions/binary_cross_entropy.py [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    Returns:
+    - bce_loss: Binary Cross-Entropy Loss.
+
+    Example Usage:
+    >>> true_labels = np.array([0, 1, 1, 0, 1])
+    >>> predicted_probs = np.array([0.2, 0.7, 0.9, 0.3, 0.8])
+    >>> binary_cross_entropy(true_labels, predicted_probs)
+    0.2529995012327421
+    """
+    # Clip predicted probabilities to avoid log(0) and log(1)
```
Comment (tianyizheng02) [inline machine_learning/loss_functions/binary_cross_entropy.py:38]: Can we have an ndarray length check like in `mean_squared_error.py`? And a doctest for that case as well

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

**Comment by THEGAMECHANGER416** [comment]
@tianyizheng02  Did all the requested changes. Hope you'll merge this now.
Thanks

**Review by tianyizheng02 — APPROVED** [review]
Thanks for your contribution!
